app/subscriber/src/features/impersonation/Impersonation.tsx

```diff
@@ -0,0 +1,176 @@
+import { Action } from 'components/action';
+import { Bar } from 'components/bar';
+import { Button } from 'components/button';
+import { PageSection } from 'components/section';
+import React from 'react';
+import {
+  FaArrowRightFromBracket,
+  FaArrowRightToBracket,
+  FaCaretLeft,
+  FaCaretRight,
+  FaMagnifyingGlass,
+} from 'react-icons/fa6';
+import { useUsers } from 'store/hooks';
+import { useProfileStore } from 'store/slices';
+import { Col, IPaged, ISubscriberUserModel, IUserFilter, IUserModel, Row, Text } from 'tno-core';
+
+import { defaultPage } from './constants';
+import * as styled from './styled';
+
+export const Impersonation = () => {
+  const { findUsers, updateUser } = useUsers();
+  const [{ profile }] = useProfileStore();
+
+  const [page, setPage] = React.useState('');
+  const [keyword, setKeyword] = React.useState('');
+  const [users, setUsers] = React.useState<IPaged<IUserModel>>(defaultPage);
+
+  const handleSearch = React.useCallback(
+    async (filter: IUserFilter) => {
+      try {
+        const page = await findUsers(filter);
+        setUsers(page);
+        setPage(`${filter.page ?? 1}`);
+      } catch {}
+    },
+    [findUsers],
+  );
+
+  const handleImpersonate = React.useCallback(
+    async (user: ISubscriberUserModel | undefined) => {
+      try {
+        if (profile) {
+          await updateUser({
+            ...profile,
+            preferences: { ...profile?.preferences, impersonate: user?.key },
+          });
+          window.location.reload();
```

Comment:
We hard refresh the page to ensure all the stored data is replaced with the impersonated user.